### ISSUE TITLE
The tab "Matériels disponibles" is empty

### ISSUE THREAD

**Issue by paulz1** [salleman33/reservation#68]
## Steps to Reproduce

1. Migrate GLPI to 10.0.2 version
2. Reinstall the last version of reservation plugin
3. Open tab ""Matériels disponibles" (Available items)

## Expected Behaviour
We should see items available for reservation and we should be able to reserve them.

## Observed Behaviour
The tab is empty. 
The available items are still visible (and reservable) via Tools->Reservations

* GLPI version: 10.0.2
* Reservation plugin version: 2.3.8
* Operating System and version : Debian 11.3
* Web server version and php version : apache2 : 2.4.54-1~deb11u1, php : 7.4.30

## Analysis
I see that some of issues are written in French, so I will use French for this issue. If it is not acceptable, I'm sorry I'll rewrite it in English.  
Now the problem.

Nous avons migré récemment notre GLPI depuis la version 9.5 sur la version 10.0.2

Au moment de la migration nous avons également mis à jour tous les plugins, dont le plugin **reservation**.

L'installation du plugin s'est bien passé, mais maintenant nous avons un petit problème. 

Au fait le volet "Matériels disponibles" est vide. Est-ce que c'est normal ?

Je joins ici un petit screenshot.

Nous pourrons toujours aller dans le menu "Outils"-> Réservations. On voit là-bas la liste du matériel disponible pour la réservation, on peut les réserver et ils vont apparaitre dans "Réservations actuelles", mais on ne peut plus le faire depuis l'onglet "Matériels disponibles".

Si j'active le mode debug sur la page de cet onglet, je vois :

1) Deux fois :
```
PHP Warning (2): Declaration of PluginReservationConfig::showForm() should be compatible with CommonDBTM::showForm($ID, array $options = Array) in /usr/share/glpi/plugins/reservation/inc/config.class.php at line 46
```
2) Puis beaucoup des erreurs :
```
SQL Error "1054": Unknown column 'glpi_reservationitems.is_deleted' in 'where clause' in query "SELECT `glpi_reservationitems`.`id`, `glpi_reservationitems`.`comment`, `glpi_computers`.`name` AS name, `glpi_computers`.`entities_id` AS entities_id, `glpi_reservationitems`.`items_id` AS items_id FROM `glpi_reservationitems` INNER JOIN `glpi_computers` ON (`glpi_reservationitems`.`itemtype` = 'Computer' AND `glpi_reservationitems`.`items_id` = `glpi_computers`.`id`) LEFT JOIN `glpi_reservations` ON (`glpi_reservationitems`.`id` = `glpi_reservations`.`reservationitems_id` AND '2022-08-04 17:33:46' < `glpi_reservations`.`end` AND '2022-08-04 20:00:00' > `glpi_reservations`.`begin`) WHERE `glpi_reservationitems`.`is_active` = '1' AND `glpi_reservationitems`.`is_deleted` = '0' AND `glpi_computers`.`is_deleted` = '0' AND `glpi_reservations`.`id` IS NULL AND ( `glpi_computers`.`entities_id` IN ('0') ) ORDER BY `glpi_computers`.`entities_id`, `glpi_computers`.`name` ASC"
```
Je ne sais pas si ca aide.

**Comment by paulz1** [comment]
J'ai oublié d'ajouter le snapshot.
![materiel_disponible](https://user-images.githubusercontent.com/6814620/182898978-e93f49d0-af47-4dcb-acdc-73a999e4cd75.png)

**Comment by salleman33** [comment]
Bonjour, 
Le plugin n'est pas encore compatible avec la version 10 de GLPI. J'ai commencé à y travailler.
Pouvez-vous tester avec la version de la branche master du dépôt git ?

**Comment by paulz1** [comment]
Merci pour la réponse :-)

Oui, je pense qu'on peut tester. 
Comment on fait pour l'installer ?
Je clone le dépôt (ou je telecharge le zip du dépôt pour eviter les fichiers de git) puis je copie le contenu à la palce du plugin actuel et je fais :
```
php bin/console glpi:plugin:desactivate <directory inside plugins>
php bin/console glpi:plugin:install --username=glpi <directory inside plugins>
php bin/console glpi:plugin:activate <directory inside plugins>
```

Est-ce qu'il y a d'autre manips à faire ?

**Comment by salleman33** [comment]
Oui, vous téléchargez les sources avec la méthode que vous souhaitez (les fichiers .git ne posent pas de problème pour glpi). 
Pour l'activation, je n'ai jamais fait en ligne de commande donc je vous fait confiance sur ces commandes.
Puis il n'y a plus qu'à croiser les doigts ;)

**Comment by paulz1** [comment]
Ok, c'est installé.
L'onglet "Matériels disponibles" est de nouveau active (on voit le matériel). :-)

Par contre nous avons un nouveau problème.
La page "Réservations actuelles" n'affiche plus les réservations perimés. Celles qui était affichées en rouge normalement.

Peut-être je crée un nouveau ticket pour cela ? Car sans doute c'est un autre problème ?

Je donne quand même quelques détails :
1. Hier, juste après l'installation, ces matériels étaient visible (affiché s en rouge) 
2. Je ne pense pas que le problème vient de la branche master (au fait nous avons déjà observé ce comportement pendant le tests (mais on n'était pas sûr, donc on n'a pas signalé).
Je pense que le problème vient sans doute au moment de changement du jour.
3. Dans l'onglet Outils -> Réservations 
Si on regarde le calendrier : on voit ces matériels.
C'est-à-dire que dans "Outils -> Réservations " je vois 13 ordinateurs (dont 10 qui n'ont pas été rendu), mais dans
"Plugins -> Réservations -> Réservations actuelles" je ne vois que 3 (ceux dont la réservation n'est pas encore périmée).

**Comment by salleman33** [comment]
il est possible que la tache qu'il faille attendre une première exécution de la tache planifiée pour que les matériels non-rendus apparaissent. 
Pouvez-vous essayer en exécutant manuellement la tache ?

**Comment by paulz1** [comment]
Oui, c'était ca. 
J'ai exécuté la tâche :  "checkReservations | Général | Surveille les réservations (plugin)" et les ordis "périmés" sont apparus.
(nous avons au fait un petit souci avec le cron glpi sur notre installation je crois)

Merci beaucoup pour votre aide.
Je pense qu'on peut fermer ce ticket. :-)
